core/vm/evm.go

```diff
@@ -220,6 +227,32 @@ func NewEVM(blockCtx BlockContext, txCtx TxContext, statedb StateDB, chainConfig
 		chainRules:  chainConfig.Rules(blockCtx.BlockNumber, blockCtx.Time),
 	}
 	evm.interpreter = NewEVMInterpreter(evm)
+
+	// If the predicate results are already available, use them.
+	if blockCtx.PredicateResults != nil {
```

Comment:
(info) As above, this should be compatible with `libevm`. The `PredicateResults` will be in the payload registered to be carried with every `Header`. If the pattern in #746 of `types.GetHeaderExtra(*Header)` is what's eventually used then that will expose it. That said, this entire method will probably be defined on the registered type, not `BlockContext` as that will be replaced with the `libevm` equivalent.